task_19/andrew.samsonov/src/main/java/ua/pp/darknsoft/CustomLinkedList.java

```diff
@@ -0,0 +1,101 @@
+package ua.pp.darknsoft;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class CustomLinkedList<T> implements Iterable<T> {
+
+  private Node<T> firstElement;
+  private Node<T> lastElement;
+
+  public void add(T element) {
+    Node<T> localNode = new Node(element, null);
+    if (firstElement == null) {
+      firstElement = lastElement = localNode;
+    } else {
+      lastElement.next = localNode;
+      lastElement = localNode;
+    }
+  }
+
+  public T poll() {
+    Node<T> localNode = firstElement;
+    firstElement = firstElement.next;
+    return localNode.item;
+  }
+
+  public T peek() {
+    return firstElement.item;
+  }
+
+  public T get(int index) {
+    return getNode(index).item;
+  }
+
+  private Node<T> getNode(int index) {
+    Node<T> localNode = firstElement;
+    int count = 0;
+
+    while (count < index) {
+      localNode = localNode.next;
+      count++;
+      if (localNode == null) {
+        throw new NoSuchElementException("No Such Element");
+      }
+    }
+    return localNode;
+  }
+
+  public void introduceCycle(int index) {
+    //зациклите список в этом методе
+    lastElement.next = getNode(index);
+  }
+
+  public boolean hasCycle() {
+    //этот метод должен определить есть ли цикл в списке
+    return lastElement.next != null;
```

Comment:
This way is definitely the best provided we have an introduceCycle method, but try to implement cycle search as we discussed in the class with fast and slow iterator as if we don't have information about the last element (we only have a reference for the first element in a list).